packages/task/src/browser/task-configurations.ts

```diff
@@ -230,6 +241,43 @@ export class TaskConfigurations implements Disposable {
         }
     }
 
+    /** Adds given task to a config file and opens the file to provide ability to edit task configuration. */
+    async configure(task: TaskConfiguration): Promise<void> {
+        const workspace = this.workspaceService.workspace;
+        if (!workspace) {
+            return;
+        }
+
+        const tasks = this.getTasks();
+        tasks.push(task);
+
+        const configFileUri = this.getConfigFileUri(workspace.uri);
+        await this.saveTasks(configFileUri, tasks);
+
+        try {
+            await open(this.openerService, new URI(configFileUri));
+        } catch (e) {
+            console.error(`Error occurred while opening: ${this.TASKFILE}.`, e);
+        }
+    }
+
+    /** Writes the tasks to a config file. Creates a config file if this one does not exist */
+    async saveTasks(configFileUri: string, tasks: TaskConfiguration[]): Promise<void> {
+        if (configFileUri && !await this.fileSystem.exists(configFileUri)) {
+            await this.fileSystem.createFile(configFileUri);
+        }
+
+        const preparedTasks = this.filterDuplicates(tasks).map(task => {
+            const { _source, $ident, ...preparedTask } = task;
+            return preparedTask;
+        });
+
+        const resource = await this.resourceProvider(new URI(configFileUri));
+        if (resource.saveContents) {
+            await resource.saveContents(JSON.stringify({ tasks: preparedTasks }, undefined, 4));
```

Comment:
We should try to preserve formatting, like here: https://github.com/theia-ide/theia/blob/3886397750b34e6cb65976578c4eb848639adfae/packages/workspace/src/browser/workspace-service.ts#L363-L364

You can use `Resource.save` and provide delta as well as full context. It knows when apply what.

Reply:
just tested, changing tasks.json and then configuring a new task via  `Run Task` breaks user formatting

Reply:
Done